app/over_react_redux/todo_client/lib/src/components/shared/list_item_component_mixin.dart

```diff
@@ -0,0 +1,121 @@
+import 'package:meta/meta.dart';
+import 'package:over_react/over_react.dart';
+
+import 'package:todo_client/src/models/base_model.dart';
+import 'package:todo_client/src/components/shared/list_item_mixin.dart';
+import 'package:todo_client/src/components/shared/material_ui.dart';
+
+mixin ListItemMixin<M extends BaseModel, T extends ListItemPropsMixin, S extends ListItemStateMixin>
+    on UiStatefulComponent2<T, S> {
+  @override
+  @mustCallSuper
+  get defaultProps => (newProps()
+    ..isEditable = false
+    ..isSelected = false
+  );
+
+  @override
+  @mustCallSuper
+  Map getDerivedStateFromProps(Map nextProps, Map prevState) {
```

Comment:
@greglittlefield-wf 

> makes your components difficult to think about.

In my opinion - this is a matter of opinion.  In some cases - like these form controls that allow users to update the values without committing the change to the store - while also ensuring that they redraw as they type - deriving state from props is a decent pattern.

The suggested alternative of using an uncontrolled component with a key that changes based on some value that updates - is equally difficult to reason about - and it will also do a full remount of the component - instead of an update... which seems... unnecessary.

I'll keep trying to think of a way to avoid this... but I don't think that the React team's opinions are infallible, either.